test/main.cc

```diff
@@ -25,6 +25,32 @@ int main(int argc, char** argv) {
   ::testing::Test::RecordProperty("TemporaryDirectory", TestEnvironment::temporaryDirectory());
   ::testing::Test::RecordProperty("RunfilesDirectory", TestEnvironment::runfilesDirectory());
 
+  if (::setenv("TEST_UDSDIR", TestEnvironment::unixDomainSocketDirectory().c_str(), 1) != 0) {
+    ::perror("Failed to temporary UDS directory.");
+    ::exit(1);
+  }
+
+  // Quick and dirty filtering and execution of --envoy_test_setup. Scripts run
+  // in this environment have access to env vars such as TEST_TMPDIR. This
+  // script can be used for situations where genrules are unsuitable for unit
+  // tests, since genrules don't have access to TEST_TMPDIR.
```

Comment:
I understand you are trying to get this done, but it would be nice if there was handled as part of a TestOptionsImpl below. Perhaps we can fix this when we convert to gflags. Can you put a TODO for @rlazarus  in?